src/modules/files-storage/api/uc/wopi.uc.ts

```diff
@@ -36,11 +40,41 @@ export class WopiUc {
 		private readonly authorizationClientAdapter: AuthorizationClientAdapter,
 		private readonly logger: Logger,
 		private readonly collaboraService: CollaboraService,
-		private readonly wopiConfig: WopiConfig
+		private readonly wopiConfig: WopiConfig,
+		private readonly em: EntityManager
 	) {
 		this.logger.setContext(WopiUc.name);
 	}
 
+	public async putFile(query: WopiAccessTokenParams, req: Request): Promise<FileRecord> {
+		const result = await this.authorizationClientAdapter.resolveToken(
+			query.access_token,
+			this.wopiConfig.WOPI_TOKEN_TTL_IN_SECONDS
+		);
+
+		const payload = WopiPayloadFactory.buildFromUnknownObject(result.payload);
+
+		const fileRecord = await this.uploadFile(payload.fileRecordId, req);
+
+		return fileRecord;
+	}
+
+	private async uploadFile(fileRecordId: EntityId, req: Request): Promise<FileRecord> {
+		const fileRecord = await RequestContext.create(this.em, async () => {
+			return await this.updateFileWithRequestData(fileRecordId, req);
```

Comment:
return this.updateFile... anstatt return await this
Allerdings erschließt sich mir hier nicht warum wir RequestContext.create ausführen. Der em ist eigentlich nur im Repo Layer bei den mongoDB Elementen und bisher noch in den api test erlaubt.

Wir haben Ausnahmen wo wir ihn aktuell nutzen, die sind aber alle architecture fails und nur drin da alternative Lösungen zu bauen zu aufwändig war.